scripts/index_walker.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+import requests
+import json
+import datetime
+from collections import defaultdict
+from itertools import chain
+from splice.environment import Environment
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.sql import select, insert, update
+from sqlalchemy import create_engine
+from splice.models import Tile, Adgroup, Country, Account, Campaign, CampaignCountry
+from tld import get_tld
+
+ARBITRARY_FUTURE = datetime.datetime.strptime('2525-01-01', "%Y-%m-%d").date()
+
+
+def main():
+    """
+    This script is going to populate Account and Campaign database structures.  It does this
+    by reading the currently deployed tile distributions (s3), where it determines the currently active
+    tile set, as well as the geo-targetting data (currently only country level) for each tile/adgroup.
+
+    The script also popultates/migrates data from the 'countries' table in the stats database.
+
+    The script will discriminate between 'active' and 'inactive' adgroups based on whether or not
+    the adgroup exists in the current distribution.  Inactive adgroups are given start/end dates
+    in campaigns that are in the *past*.  Active adgroups are placed in campaigns that start on their
+    adgroup creation date and end at some far distant future date.
+
+    Campaign objects are considered unique by grouping together the following keys in the adgroup:
+    * the TLD+1 of the target_url
+    * the locale of the adgroup
+    * the channel of the adgroup
+    * the 'active' flag (determined as explained above) of the adgroup
+
+    One campaign row will be assigned for each unique campaign detected.
+
+    The script will populate the adgroup.campaign_id with the campaign that the adgroup fits into.
+
+    Account objects are considered unique by considering the TLD+1 of the campaign.  All campaigns that
```

Comment:
For content tile partners `TLD+1` might not be sufficient. For example, Conde Nast has 21 sub-brands, which all equipped with different domains.